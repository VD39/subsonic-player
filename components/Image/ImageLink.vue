<script setup lang="ts">
import PreloadImage from '@/components/Image/PreloadImage.vue';

defineProps<{
  to: string;
  title: string;
  image: string;
}>();
</script>

<template>
  <figure :class="$style.figure">
    <NuxtLink ref="link" :to="to" :title="`Go to ${title}`">
      <PreloadImage :image="image" :alt="title" />
    </NuxtLink>
  </figure>
</template>

<style module>
.figure {
  &:hover {
    opacity: 0.8;
  }
}
</style>
