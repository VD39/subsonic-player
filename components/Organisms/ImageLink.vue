<script setup lang="ts">
import PreloadImage from '@/components/Molecules/PreloadImage.vue';

defineProps<{
  image: Image;
  title: string;
  to: ToProp;
}>();
</script>

<template>
  <figure>
    <NuxtLink draggable="false" :title :to>
      <PreloadImage :alt="title" :image />
    </NuxtLink>
  </figure>
</template>
