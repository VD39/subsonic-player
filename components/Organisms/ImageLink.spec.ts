import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import ImageLink from './ImageLink.vue';

function factory(props = {}) {
  return mount(ImageLink, {
    props: {
      image: 'image',
      title: 'title',
      to: '/to',
      ...props,
    },
  });
}

describe('ImageLink', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
