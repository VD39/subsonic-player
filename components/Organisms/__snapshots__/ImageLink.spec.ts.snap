// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageLink > matches the snapshot 1`] = `
"<figure><a title="title">
    <div class="centerAll preloadImage"><span class="imageLoader"><span class="visuallyHidden">Loading image</span></span><img alt="title" class="image" src="image" style="display: none;"></div>
  </a></figure>"
`;
